api/external_resources/models.py

```diff
@@ -0,0 +1,102 @@
+from django.db import models
+from django_lifecycle import (
+    AFTER_SAVE,
+    BEFORE_DELETE,
+    LifecycleModelMixin,
+    hook,
+)
+
+from features.models import Feature, FeatureState
+from integrations.github.tasks import call_github_app_webhook_for_feature_state
+from webhooks.webhooks import WebhookEventType
+
+
+class ExternalResources(LifecycleModelMixin, models.Model):
+    url = models.URLField()
+    type = models.TextField()
+    status = models.TextField(null=True)
+    feature = models.ForeignKey(
+        Feature,
+        related_name="features",
+        on_delete=models.CASCADE,
+    )
+
+    @hook(AFTER_SAVE)
+    def trigger_feature_external_resource_added_hook(self):
+        if hasattr(self.feature.project.organisation, "github_config"):
+            github_configuration = self.feature.project.organisation.github_config
+
+            feature_states = FeatureState.objects.filter(feature_id=self.feature_id)
+            feature_data = {
+                "id": self.feature_id,
+                "name": self.feature.name,
+                "feature_states": [],
+            }
+            feature_data["installation_id"] = github_configuration.installation_id
+            feature_data["organisation_id"] = github_configuration.organisation.id
+
+            for feature_state in feature_states:
+                feature_env_data = {}
+                if feature_state.feature_state_value.string_value is not None:
```

Comment:
Can we move this type checking to `github` module?

Reply:
Moved